lib/rust/metamodel/src/meta/mod.rs

```diff
@@ -0,0 +1,388 @@
+    pub fn from_camel_case(s: &str) -> Self {
+        Self::from_pascal_case(s)
+    }
+
+    /// Parse an identifier expected to be in PascalCase.
+    pub fn from_pascal_case(s: &str) -> Self {
+        let mut segments = vec![];
+        let mut current = String::new();
+        for c in s.chars() {
+            if c.is_ascii_uppercase() && !current.is_empty() {
+                segments.push(std::mem::take(&mut current));
+            }
+            current.push(c.to_ascii_lowercase());
+        }
+        segments.push(current);
+        Self::new(segments)
+    }
+
+    /// Append another `Identifier` to the end of `self`; when rendered, the boundary between the
+    /// old and new components will be indicated in a manner determined by the naming convention
+    /// chosen at rendering time.
+    pub fn append(&mut self, other: Self) {
+        self.segments.extend(other.segments)
+    }
+}
+
+
+// === Type Names ===
+
+/// The name of a type, e.g. a `struct` in Rust or a `class` in Java.
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct TypeName(Identifier);
+
+impl std::fmt::Display for TypeName {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str(&self.0.to_pascal_case())
+    }
+}
+
+impl TypeName {
+    /// Parse from PascalCase.
+    pub fn from_pascal_case(s: &str) -> Self {
+        Self(Identifier::from_pascal_case(s))
+    }
+    /// Render in PascalCase.
+    pub fn to_pascal_case(&self) -> String {
+        self.0.to_pascal_case()
+    }
+    /// Append another `TypeName` to the end of `self`. See `Identifier::append`.
+    pub fn append(&mut self, other: Self) {
+        self.0.append(other.0)
+    }
+}
+
+
+// === Field Names ===
+
+/// The name of a field, e.g. the data members of a Rust struct or Java class.
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash, Default)]
+pub struct FieldName(Identifier);
+
+impl std::fmt::Display for FieldName {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str(&self.0.to_camel_case())
+    }
+}
+
+impl FieldName {
+    /// Parse from snake_case.
+    pub fn from_snake_case(s: &str) -> Self {
+        Self(Identifier::from_snake_case(s))
+    }
+    /// Render in camelCase.
+    pub fn to_camel_case(&self) -> Option<String> {
+        match self.0.to_camel_case() {
+            ident if ident.is_empty() => None,
+            ident => Some(ident),
+        }
+    }
+    /// Append another `FieldName` to the end of `self`. See `Identifier::append`.
+    pub fn append(&mut self, other: Self) {
+        self.0.append(other.0)
+    }
+}
+
+
+
+// ===========================
+// === System of Datatypes ===
+// ===========================
+
+/// A system of `Type`s.
+#[derive(Debug, Default, Clone, Index, IndexMut)]
+pub struct TypeGraph {
+    #[index]
+    #[index_mut]
+    #[allow(missing_docs)]
+    pub types: VecMap<Type>,
```

Comment:
These docs do not explain anything. What is "system" of Types? What is the purpose of this struct? It seems that this is an important struct, but there is no explanation for it.